backend/src/chat_server/events/report_events.ts

```diff
@@ -9,13 +9,19 @@ export function registerReportEvents(socket: SocketIO.Socket) {
       if (!(await isAdmin(this.localSocketState[socket.id].username))) {
         throw "This user isn't Admin";
       }
-
-      let userState = await UserManager.banUser(username);
-      this.emitBanMessage(username).catch((e) => console.error(e));
-      return this.io.to(socket.id).emit("report_user_resp", {
-        status: -1,
-        response: `Cấm chat ${username} thành công!`,
-      });
+      if (isAdmin(username)) {
+        return this.io.to(socket.id).emit("report_user_resp", {
+          status: -1,
+          response: `Bạn không thể cấm chat admin ${username}.`,
+        });
+      } else {
+        let userState = await UserManager.banUser(username);
+        this.emitBanMessage(username).catch((e) => console.error(e));
+        return this.io.to(socket.id).emit("report_user_resp", {
+          status: -1,
+          response: `Cấm chat ${username} thành công!`,
+        });
+      }
```

Comment:
```suggestion
      if (isAdmin(username)) {
        return this.io.to(socket.id).emit("report_user_resp", {
          status: -1,
          response: `Bạn không thể cấm chat admin ${username}.`,
        });
      }
      let userState = await UserManager.banUser(username);
      this.emitBanMessage(username).catch((e) => console.error(e));
      return this.io.to(socket.id).emit("report_user_resp", {
        status: -1,
        response: `Cấm chat ${username} thành công!`,
      });
```
I personally think making the codeblocks less nested would help with readability a lot :D. Keep in mind that we already have to deal with "callback hell" sometimes so making the code less nested whenever you can would definitely help!